### PR TITLE
Fix ResultSet __eq__ to handle nans

### PR DESCRIPTION
**Pull request by gaffney2010**
Fixes https://github.com/Axelrod-Python/Axelrod/issues/1286

### CODE REVIEW

File: axelrod/result_set.py [hunk 1]
```diff
@@ -609,6 +610,17 @@ def __eq__(self, other):
             other : axelrod.ResultSet
                 Another results set against which to check equality
         """
+        def list_equal(v1: List[float], v2: List[float]) -> bool:
```
Comment (marcharper) [inline axelrod/result_set.py]: Can we say more specifically that nans are considered equal, and rename the function to something like `list_equal_with_nans` or something like that?

File: axelrod/result_set.py [hunk 2]
```diff
@@ -609,6 +610,17 @@ def __eq__(self, other):
...
+            """Matches lists, accounting for NaNs."""
+            if len(v1) != len(v2):
+                return False
+            for i1, i2 in zip(v1, v2):
+                if np.isnan(i1) and np.isnan(i2):
+                    continue
+                if i1 != i2:
+                    return False
```
Comment (marcharper) [inline axelrod/result_set.py]: This line is the one not covered by a test

File: axelrod/result_set.py [hunk 3]
```diff
@@ -722,7 +751,8 @@ def summarise(self):
 
         summary_data = []
         for rank, i in enumerate(self.ranking):
-            data = list(summary_measures[i]) + state_prob[i] + state_to_C_prob[i]
+            data = list(summary_measures[i]) + state_prob[i] + state_to_C_prob[
+                i]
```
Comment (marcharper) [inline axelrod/result_set.py]: Can you run black on the final commit? Some of the line wraps seem odd.

File: axelrod/tests/unit/test_resultset.py [hunk 4]
```diff
@@ -190,6 +190,52 @@ def test_init(self):
         self.assertEqual(rs.players, self.players)
         self.assertEqual(rs.num_players, len(self.players))
 
+    def _clear_matrix(self, matrix):
+        for i in range(len(matrix)):
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_resultset.py]: FYI, this is sometimes considered an anti pattern and the following is preferred:

```
for i, row in enumerate(matrix):
    for j, _ in enumerate(row):
         matrix[i, j] = 0
```

File: axelrod/tests/unit/test_resultset.py [hunk 5]
```diff
@@ -190,6 +190,52 @@ def test_init(self):
...
+            for j in range(len(matrix[i])):
+                matrix[i][j] = 0
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_resultset.py]:
```suggestion
    def _clear_matrix(self, matrix):
        for i, row in enumerate(matrix):
            for j, _ in enumerate(row):
                 matrix[i, j] = 0
```

**Comment by drvinceknight** [comment]
Looks great TJ, one very minor suggestion from me.

File: axelrod/result_set.py (hunk 3 above)
Comment (gaffney2010) [reply axelrod/result_set.py]: Yep.  I'm using a new IDE, but just added black, so it should be better going forward.

**Pull request merged**